### PR TITLE
Patch function "Display Name" into admin page

### LINKED ISSUE #2100: Function "Display Name" administration page
## Before you open your issue
- [X] Check if no issue or pull-request for this already exists.
- [X] Check [documentation](https://mailu.io/master/) and [FAQ](https://mailu.io/master/faq.html). (Tip, use the search function on the documentation page)
- [X] You understand `Mailu` is made by volunteers in their **free time** — be conscise, civil and accept that delays can occur.
- [X] The title of the issue should be short and simple. It should contain specific terms related to the actual issue. Be specific while writing the title.

### Environment
 - [X] docker-compose

### Versions

`master hash 18865bf Tue Dec 21 13:08:48 2021`

## Description
The [master Display Name](https://mailu.io/master/webadministration.html#display-name) documentation describes:

> On the settings page the displayed name can be changed of the logged in user. This display name is only used within the web administration interface.

At which locations should the `Displayed name` shown? When I login, I see the email address in the left bar and not display name.

image

## Replication Steps
1. Login to Administration page
2. Change `Displayed name` and click `Save`
3. The user icon shows the email address and not filled in Displayed Name.

Thanks!

### PR DESCRIPTION
**Pull request by 0pc0deFR**
## What type of PR?

bug-fix

## What does this PR do?

This PR is bug-fix for #2100 

### Related issue(s)
- Mention an issue like: #2100
- Auto close an issue like: closes #2100

### CODE REVIEW

File: core/admin/mailu/ui/templates/sidebar.html [hunk 1]
```diff
@@ -5,7 +5,11 @@
       <div class="div-circle elevation-2"><i class="fa fa-user text-lg text-dark"></i></div>
     </div>
     <div class="info">
+      {%- if current_user.displayed_name %}
+      <a href="{{ url_for('.user_settings') }}" class="d-block">{{ current_user.displayed_name }}</a>
+      {% else %}
       <a href="{{ url_for('.user_settings') }}" class="d-block">{{ current_user }}</a>
+      {%- endif %}
```
Comment (ghostwheel42) [inline core/admin/mailu/ui/templates/sidebar.html:12]: You should be able to use
`<a href="{{ url_for('.user_settings') }}" class="d-block">{{ current_user.displayed_name or current_user }}</a>`
instead.

Comment (0pc0deFR) [reply core/admin/mailu/ui/templates/sidebar.html]: That's a good point. I fixed this code.

File: core/admin/mailu/ui/templates/sidebar.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
       <div class="div-circle elevation-2"><i class="fa fa-user text-lg text-dark"></i></div>
     </div>
     <div class="info">
-      <a href="{{ url_for('.user_settings') }}" class="d-block">{{ current_user }}</a>
+      <a href="{{ url_for('.user_settings') }}" class="d-block">{{ current_user.displayed_name or current_user }}</a>
```
Comment (ghostwheel42) [inline core/admin/mailu/ui/templates/sidebar.html:8]: Sorry, but I think this also should be escaped.
I didn't test this but Jinja has an escape filter called "e", so this should work:
`current_user.displayed_name|e or current_user|e`
Could you add a title attribute to the div (class user-panel) above containing the email-address?
`<div class="user-panel mt-3 pb-3 mb-3 d-flex" title="{{ current_user|e }}">`

Comment (0pc0deFR) [reply core/admin/mailu/ui/templates/sidebar.html:8]: I thnk jinja autoescape inputs if isn't disabled : https://flask.palletsprojects.com/en/1.1.x/templating/#controlling-autoescaping

I add title attribute on user-panel div.

**Comment by ghostwheel42** [comment]
Auto-Quoting is enabled. So this looks good.
While testing I've seen that it is possible to input just a space as displayed name.
Maybe we should better check/normalize all input fields.
But not in this PR.

**Review by ghostwheel42 — APPROVED** [review]
lgtm

**Review by nextgens — APPROVED** [review]

**Pull request merged**